### PR TITLE
Add vendor and allow offline builds

### PR DESCRIPTION
**Pull request by tsmetana**
In order to be able to build the project in offline environment (no internet) we can't do the dep ensure step. This means the vendor directory has to be present and we only need to call go build. This required also changes to the spec file, hece the two commits in the PR.

### CODE REVIEW

**Comment by tsmetana** [comment]
@jsafrane PTAL. The spec file is a bit hackish but the assumptions that cloud-provider-openstack makes about its environment make it somewhat more complicated to package.

File: cloud-provider-openstack.spec [hunk 1]
```diff
@@ -74,35 +74,28 @@ Summary: Provisioner for OpenStack Manila
 Provisions volumes using OpenStack Manila API.
 
 %prep
+# cloud-provider-openstack assumes it's compiled in a Go workspace
+# Let's create one
 %if 0%{do_prep}
-%setup -q
+%setup -n cloud-provider-openstack
+mkdir -p src/k8s.io
+for dep in $(ls vendor); do
```
Comment (jsafrane) [inline cloud-provider-openstack.spec]: This is awfully complicated. All our other packages use this to create GOPATH:

```
mkdir -p src/%{provider}.%{provider_tld}/%{project}
ln -s ../../../ src/%{import_path}
export GOPATH=$(pwd):$(pwd)/repo-infra/vendor:%{gopath}
```

http://pkgs.devel.redhat.com/cgit/rpms/csi-attacher/tree/csi-attacher.spec?h=rhaos-3.10-rhel-7#n53


You should not unroll vendor/ into src/.

Comment (tsmetana) [reply cloud-provider-openstack.spec]: This whole think is actually a side-effect of the testing builds: "make manila-provisioner" expect to be run from a "proper" go workspace so i created one so i could fiddle with things after %prep...

Will simplify it. Thanks.

**Comment by tsmetana** [comment]
@jsafrane I updated the spec file: it works with much less fuss: I build the gofile directly because that's what the Makefile target would do and I don't want to diverge from the upstream.

**Pull request merged**